tests/test_properties.py

```diff
@@ -40,6 +40,108 @@ def eval(self, x):
         return self.a * x + self.b
 
 
+def g(x):
+    """Non-linear model, Example 5.1 from Evensen 2019"""
+    x1 = x[0]
+    return x1 + 0.2 * x1**3
+
+
+@pytest.mark.parametrize("N", [100, 200])
+def test_that_projection_is_better_for_nonlinear_forward_model_big_N_small_p(N):
+    # For non-linear forward model, g, and m<N,
+    # Eq.27 should provide a better update than Eq. 28.
+    # "Better" in terms of a better optimum, given by loss in Eq. 10
+    # We here solve using both Equations 27 and 28 (with/without projection)
+    # and then evaluate the loss function.
+
+    def loss_function(xj, xj_prior, dj, Cxx, Cdd, g):
```

Comment:
Just a question, but do you think we would get the same results using Eq. 18 as loss function instead of Eq. 10?

Reply:
Yes, I think so. The loss in Eq. 18 is equivalent to Eq. 10 but constrained to searching for a solution in the ensemble subspace through only considering updates of the form Eq. 16. Since the solutions (with and without projection) both are of the form Eq. 16, evaluating them with Eq. 10 should be equivalent to directly evaluating Eq. 18.